### ISSUE TITLE
Fresh LoRA training starts from a randomly perturbed model, not from the base model

### ISSUE THREAD

**Issue by azrahello** [mflux-community/mflux#739]
`LoRALinear.__init__` draws both factors from `U(-1/sqrt(in), 1/sqrt(in))`, so `ΔW = A @ B`
  is non-zero before the first gradient. Every patched layer already perturbs the base model
  at step 0, and the optimizer's first steps go into undoing it.

  The LoRA paper prescribes the opposite ("a random Gaussian initialization for A and zero for
  B, so ΔW = BA is zero at the beginning of training", Hu et al. 2021, §4.1), and PEFT,
  diffusers and ai-toolkit all zero-init B.

  1. Start a fresh training run (no resume, no preloaded adapter) with any model.
  2. Look at the step-0 preview, or at `lora_B` inside `0000000_checkpoint.zip`.
  3. The preview is not the base model, and `lora_B` is not zero.

  Measured on Krea 2 Turbo, 264 targets at rank 16, seed fixed:
  
  | | step-0 checkpoint |
  |---|---|
  | layers with non-zero `lora_B` | 264 / 264 |
  | `\|\|ΔW\|\| / \|\|W\|\|` per layer | ~3% |

  Three consequences, all at step 0, before any training has happened:
  
  **The preview is not the base model.** It is the base model plus a rank-16 perturbation on
  every patched layer.

  **It shows a visible grid artifact.** The perturbation reaches the image as a regular pattern
  whose period is exactly 8 px — the VAE's `spatial_scale`, each latent value expanding into an
  8x8 block. Spectral peak at that frequency, relative to the median of the spectrum:

  | run | peak |
  |---|---|
  | 32 targets (includes `first` and `last.linear`) | 84.5 |
  | same, with `first` and `last.linear` removed | 33.7 |
  | `lora_B` zero-initialized | 29.4 |

  Dropping just those two targets removes almost all of it, which is consistent with where the
  spatial grid is explicit: `first` packs the latent into tokens and `last.linear` unpacks it,
  so a fixed perturbation there is applied identically to every cell.

  **It is not reproducible.** Anything that consumes the global RNG before the injection changes
  the starting point. Loading an assistant training adapter constructs 256 `LoRALinear` objects,
  each drawing two random factors that are then overwritten from file — 512 draws consumed, so
  the user's own factors differ. Two step-0 checkpoints from the same seed, one with the adapter
  loaded and one without, shared **0 of 528 identical tensors**. The same seed also produced a
  different subject in the preview: with a prompt that does not specify it ("wearing a bikini"),
  one run gave a floral pattern and the other a plain one.

  With `lora_B` zeroed, two different code paths (one loading an assistant adapter, one not)
  produced a **pixel-identical** step-0 preview, RMSE 0.000 — the starting point stops depending
  on anything but the base model.

  To check without a training run:

  ```python
  import mlx.core as mx
  from mlx import nn
  from mflux.models.common.lora.layer.linear_lora_layer import LoRALinear
  
  linear = nn.Linear(8, 4, bias=False)
  x = mx.random.uniform(shape=(2, 8))
  wrapped = LoRALinear.from_linear(linear, r=4, scale=1.0)
  print(mx.array_equal(wrapped(x), linear(x)).item())   # False; should be True

  Fix: initialize lora_B to zeros. lora_A stays random, so the layer is not dead — it gets
  gradient as soon as B moves. Loading a saved adapter is unaffected: lora_loader.py:361
  rejects any adapter missing either factor, and all three apply paths overwrite both.
